sys/include/net/gnrc/ndp2.h

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_ndp2   IPv6 Neighbor discovery (v2)
+ * @ingroup     net_gnrc_ipv6
+ * @brief       Provides build and send functions for neighbor discovery packets
+ * @{
+ *
+ * @file
+ * @brief       GNRC-specific neighbor discovery definitions
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef GNRC_NDP2_H
+#define GNRC_NDP2_H
+
+#include <stdint.h>
+
+#include "kernel_types.h"
+#include "net/gnrc/pkt.h"
+#include "net/gnrc/ipv6/netif.h"
+#include "net/ipv6/addr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   @ref net_gnrc_nettype to send NDP packets to
+ */
+#ifndef GNRC_NETTYPE_NDP2
+# if    defined(MODULE_GNRC_IPV6) || DOXYGEN
+#  define GNRC_NETTYPE_NDP2 (GNRC_NETTYPE_IPV6)
+# else
+#  define GNRC_NETTYPE_NDP2 (GNRC_NETTYPE_UNDEF)
+# endif
+#endif  /* GNRC_NETTYPE_NDP2 */
+
+/**
+ * @brief   Builds a neighbor solicitation message for sending.
+ *
+ * @pre `(tgt != NULL) && !ipv6_addr_is_multicast(tgt)`
+ *
+ * @see [RFC 4861, section 4.3](https://tools.ietf.org/html/rfc4861#section-4.3")
+ *
+ * @param[in] tgt       The target address of the neighbor solicitation.
+ *                      May not be NULL and **MUST NOT** be multicast.
+ * @param[in] options   Options to append to the neighbor solicitation.
+ *                      May be NULL for none.
+ *
+ * @return  The resulting ICMPv6 packet on success.
+ * @return  NULL, if packet buffer is full.
+ */
+gnrc_pktsnip_t *gnrc_ndp2_nbr_sol_build(const ipv6_addr_t *tgt,
+                                        gnrc_pktsnip_t *options);
+
+/**
+ * @brief   Builds a neighbor advertisement message for sending.
+ *
+ * @pre `(tgt != NULL) && !ipv6_addr_is_multicast(tgt)`
+ *
+ * @see [RFC 4861, section 4.4](https://tools.ietf.org/html/rfc4861#section-4.4")
+ *
+ * @param[in] tgt       For solicited advertisements, the Target Address field
+ *                      in the neighbor solicitaton.
+ *                      For and unsolicited advertisement, the address whose
+ *                      link-layer address has changed.
+ *                      May not be NULL and **MUST NOT** be multicast.
+ * @param[in] flags     Neighbor advertisement flags:
```

Comment:
Hm, appears to be a problem in my browser then. It's shifted wrongly here. There's no tab or thin whitespace there, right?